staker/block_validator.go

```diff
@@ -170,11 +170,21 @@ func (c *BlockValidatorConfig) Validate() error {
 			}
 		}
 	}
+	if c.Dangerous.Revalidation.EndBlock > 0 && c.Dangerous.Revalidation.EndBlock < c.Dangerous.Revalidation.StartBlock {
+		return fmt.Errorf("revalidation end block %d is before start block %d", c.Dangerous.Revalidation.EndBlock, c.Dangerous.Revalidation.StartBlock)
+	}
 	return nil
 }
 
 type BlockValidatorDangerousConfig struct {
-	ResetBlockValidation bool `koanf:"reset-block-validation"`
+	ResetBlockValidation bool               `koanf:"reset-block-validation"`
+	Revalidation         RevalidationConfig `koanf:"revalidation"`
+}
+
+type RevalidationConfig struct {
+	StartBlock            uint64 `koanf:"start-block"`
+	EndBlock              uint64 `koanf:"end-block"`
+	QuitAfterRevalidation bool   `koanf:"quit-after-revalidation"`
```

Comment:
Does it ever make sense to continue running after revalidating blocks?  How about simply quitting if end-block non-zero?

Reply:
We only quit the validator after the end block, QuitAfterRevalidation is for quitting the node itself (It was decided that we want the option of allowing to leave the node running even after revalidating the blocks so that different rpc calls can be tried out once the revalidating is done. )